architecture/2024-08-dashboard-component-threat-model.md

```diff
@@ -0,0 +1,299 @@
+# Radius Dashboard Component Threat Model
+
+- **Author**: @nithyatsu
+
+## Overview
+
+This document provides a threat model for the Radius Dashboard component. It identifies potential security threats to this part of Radius and suggests possible mitigations. The document includes an analysis of the system, its assets, identified threats, and recommended security measures to protect the system.
+
+The Radius Dashboard component provides the frontend experience for Radius. 
+It provides visual and textual representation of user's applications, environments and recipes.
+
+## Terms and Definitions
+
+| Term                  | Definition      |
+| --------------------- | ----------------------------- |
+| mTLS | Mutual Transport Layer Security (mTLS) allows two parties to authenticate each other during the initial connection of an SSL/TLS handshake. |
+| UCPD | Universal Control Plane Daemon for Radius |
+| DoS | Denial of Service |
+| SPA | Single Page Application (SPA): Typically a React application that contains the logic for most user interactions within the React components. SPAs do not require reloading the page from the server during user interactions. |
```

Comment:
This feels a little bit clumsy. SPA != React. There are other SPA frameworks. Might be simpler for readers if we just say what React is and provide a link.